rasa/core/channels/botframework.py

```diff
@@ -214,14 +214,19 @@ def blueprint(
                         postdata["recipient"],
                         postdata["serviceUrl"],
                     )
-
+                    
                     user_msg = UserMessage(
-                        postdata["text"],
-                        out_channel,
-                        postdata["from"]["id"],
+                        text=postdata.get("text", ""),
+                        output_channel=out_channel,
+                        sender_id=postdata["from"]["id"],
                         input_channel=self.name(),
                         metadata=metadata,
                     )
+                    if postdata.get('attachments'):
+                        user_msg.metadata = {
```

Comment:
this would make the `self.get_metadata()` call in l. 206 useless. can we instead _add_ the `attachments` key to `metadata` if it's not `None`?

Reply:
You're right! I moved the attachements detection in a function, let me know what you think about this